zebROS_ws/src/behaviors/src/path_intake_actionlib_server.cpp

```diff
@@ -0,0 +1,165 @@
+#include <ros/ros.h>
+#include <actionlib/server/simple_action_server.h>
+
+#include <behavior_actions/GamePiecePickup.h>
+#include <nav_msgs/Path.h>
+#include <behavior_actions/IntakeAction.h>
+#include <path_follower_msgs/PathAction.h>
+
+// This action lib server will first generate a path with game_piece_path_gen, then run that path using PathAction, then intake a game piece using IntakeAction.
+
+void waitForActionlibServer(T &action_client, double timeout, const std::string &activity);
+
+class PathIntakeAction{
+  private:
+    ros::NodeHandle nh_;
+    actionlib::SimpleActionServer<behaviors_actions::IntakeAction> as_;
+    std::string action_name_;
+    behaviors_actions::PathIntakeFeedback feedback_;
+    behaviors_actions::PathIntakeResult result_;
+
+  public:
+    PathIntakeAction(std::string name):
+      // boost bind is like a callback
```

Comment:
Yep, more specifically, it combines a particular instance of an object plus a member function of that object into a single function-like thing which can be passed to code which expects just a free (non-class member) function as an argument.